### PR TITLE
Fix JsonlStreamParser on system.init plugins schema drift

### LINKED ISSUE #42: JsonlStreamParser fails on system.init: 'plugins' is now an array of {name, path} objects, not strings
## Repo / version

- **File:** [`src/ClaudeAgentSdkProvider/Parsers/JsonlStreamParser.cs:41`](src/ClaudeAgentSdkProvider/Parsers/JsonlStreamParser.cs#L41)
- **Affected version:** 1.0.27 (commit `1a48360`), also reproduces on `6ec7b7b`
- **`claude-agent-sdk` CLI version observed:** `2.0.55`

## Symptom

Every `system.init` line from `claude-agent-sdk` v2.0.55+ fails to deserialize. The parser logs a warning, skips the line, and **`SystemInitMessage` is never relayed** to subscribers — so callers can never observe `SessionId`, `Model`, `Tools`, `McpServers`, `SlashCommands`, or `Agents` from the SDK's own init payload.

Observed log (the warning is emitted twice — once per warning site in the parser):

```
[18:30:21 WRN] Failed to parse JSONL line: {"type":"system","subtype":"init",...}
System.Text.Json.JsonException: The JSON value could not be converted to System.String.
Path: $.plugins[0] | LineNumber: 0 | BytePositionInLine: 8911.
 ---> System.InvalidOperationException: Cannot get the value of a token type 'StartObject' as a string.
   at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(...)
   ...
   at AchieveAi.LmDotnetTools.ClaudeAgentSdkProvider.Parsers.JsonlStreamParser.ParseLine(String jsonLine)
     in src/ClaudeAgentSdkProvider/Parsers/JsonlStreamParser.cs:line 41
```

## Root cause

`claude-agent-sdk` CLI ≥ 2.0.55 emits `plugins` as an **array of objects**, not strings:

```jsonc
{
  "type": "system",
  "subtype": "init",
  "session_id": "c76e26f4-3a82-4908-a1d7-55286fdfc3be",
  "model": "claude-opus-4-7",
  "...": "...",
  "plugins": [
    { "name": "example-skills",  "path": "C:\Users\...\anthropic-agent-skills\\" },
    { "name": "document-skills", "path": "C:\Users\...\anthropic-agent-skills\\" },
    { "name": "clean-builds",    "path": "B:\sources\claude_plugins\clean-builds" },
    { "name": "debugging",       "path": "B:\sources\claude_plugins\debugging" },
    { "name": "code-reviewer",   "path": "B:\sources\claude_plugins\code-reviewer" }
  ]
}
```

The C# model the parser binds to (the DTO behind `SystemInitMessage.Plugins`) is typed as `List<string>` / `string[]`. `System.Text.Json` throws `InvalidOperationException: Cannot get the value of a token type 'StartObject' as a string.` on `$.plugins[0]`, and `ParseLine` swallows the whole line.

## Reproduction

1. Install `claude-agent-sdk` CLI ≥ 2.0.55.
2. Have at least one Claude Code plugin installed (e.g. `code-reviewer` under `~/.claude/plugins/...`) so the CLI emits a non-empty `plugins` array.
3. Construct any `ClaudeAgentLoop` and call `RunAsync` / send a single user message.
4. Observe the `Failed to parse JSONL line` warning the moment the CLI emits its init frame; no `SystemInitMessage` is produced downstream.

## Impact

- The full init payload — `tools`, `mcp_servers` (with per-server `status`), `slash_commands`, `agents`, `permissionMode`, `model` — is lost from the public message stream. Useful diagnostic info that's now invisible to subscribers.
- Any caller that observes `SessionId` via `SystemInitMessage` relay (rather than the newer `ClaudeAgentLoop.CurrentSessionId` property) silently degrades — re-review / resume flows that key off the relayed event are blocked.
- The warning fires on **every** run, polluting Debug-level logs with a 9 KB stack trace per call.

## Suggested fix

Change the `Plugins` property on the init-message DTO from `List<string>` to a structured shape:

```csharp
public sealed record PluginRef
{
    [JsonPropertyName("name")] public string Name { get; init; } = "";
    [JsonPropertyName("path")] public string Path { get; init; } = "";
}

// On SystemInitMessage (or the DTO the parser binds):
[JsonPropertyName("plugins")]
public List<PluginRef>? Plugins { get; init; }
```

If backwards compatibility with older CLI versions that emitted bare strings is desired, a small `JsonConverter` that accepts either `string` *or* `{ name, path }` would handle both. But `claude-agent-sdk` has shipped the object shape long enough that a clean break is probably fine.

## Related

- Downstream of the same payload that previously surfaced the `SystemInitMessage` relay fix (`ba2d474`) and the `--setting-sources ""` issue (the omit-when-default fix). Those are working correctly in 1.0.27 — this is new schema-drift from the CLI bump.
- A consumer-side log line confirming end-to-end is in [revobot dotnet/](https://github.com/gautam-achieveai/revobot) at commit `13d8ad0`; happy to share the full failing JSONL payload (~9.8 KB) if useful — just say the word and I'll attach it as a gist.

### PR DESCRIPTION
**Pull request by gautam-achieveai**
[Gautam's Dev bot]

## Summary

Fixes the `JsonlStreamParser` failure on `system.init` frames emitted by `claude-agent-sdk` CLI v2.0.55+, where `plugins` changed from `string[]` to an array of `{name, path}` objects. Previously this drift caused the entire init frame (including `session_id` and `model`) to be silently dropped, breaking the `--resume` round-trip.

- Add `PluginRef` record (`Name`, `Path`) and switch `SystemInitEvent.Plugins` to `List<PluginRef>?`.
- No tolerant `JsonConverter` — clean swap (no in-repo consumers of the old shape).
- Add `JsonlStreamParser` tests covering object-form plugins, empty plugins, and a `SessionId` round-trip that locks the resume path.

## Test Plan

- [x] `dotnet test tests/ClaudeAgentSdkProvider.Tests/ClaudeAgentSdkProvider.Tests.csproj` — 29/29 passing locally (3 new)
- [x] Husky `format-verify` pre-commit hook clean

Closes #42

### CODE REVIEW

**Comment by gautam-achieveai** [comment]
[Gautam's Dev bot] **Preview environments**

| Service | URL | Status |
|---|---|---|
| lmdotnettools | http://pr-43.lmdotnettools.localhost/ | ✅ Healthy |

**Review by gautam-achieveai — COMMENTED** [review]
**[Gautam's review bot]** — Deep review of PR #43 (iteration 1)

### Context
Fixes #42 — `claude-agent-sdk` CLI v2.0.55+ changed `system.init.plugins` from `string[]` to `{name, path}[]`. The old `List<string>` typing caused `System.Text.Json` to throw on `StartObject`, and the parser's catch block silently dropped the entire `system.init` frame — losing `SessionId`, `Model`, and all other init data. This broke the `--resume` flow.

### Review Agents Used
- `code-reviewer:test-coverage-review` — tests well-targeted, adequate coverage
- `code-reviewer:architecture-review` — layer boundaries clean, `PluginRef` placement correct

### What Looks Good
- **Focused fix:** Minimal change that solves the exact problem
- **Correct placement:** `PluginRef` colocated with `SystemInitEvent`, consistent with `McpServerStatus` pattern
- **Good tests:** 3 tests covering object-form plugins, empty array, and SessionId preservation. All pass ✅
- **Clean layering:** `PluginRef` stays in event model layer; `SystemInitMessage` (public `IMessage`) unaffected

### Findings

**[MEDIUM] Backward compat with pre-v2.0.55 CLI** — Type change from `List<string>?` → `List<PluginRef>?` means older CLIs emitting string-form plugins will now fail identically (silent `system.init` drop). Issue #42 explicitly accepts this clean break, and no downstream code reads `Plugins`, so practical risk is low. A polymorphic `JsonConverter` accepting both formats would be more resilient — the codebase has prior art in `LmCore/Utils/Union.cs`. Also: `Agents` and `Skills` are still `List<string>?` — if the CLI evolves those to objects too, same pattern recurs.

**[LOW] Future drift risk for `Skills` and `Agents`** — Still `List<string>?`. Consider a tracking issue if CLI schema changes are frequent.

### Build & Tests
- `dotnet build` — ✅ 0 warnings, 0 errors
- `dotnet test --filter "SystemInit"` — ✅ 3/3 passed

### Verdict: **APPROVE** ✅
Clean, well-tested fix for a real production issue. The backward compat tradeoff is explicitly documented in #42 and is reasonable.

File: src/ClaudeAgentSdkProvider/Models/JsonlEvents/SystemInitEvent.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public record SystemInitEvent : JsonlEventBase
     public List<string>? Skills { get; init; }
 
     [JsonPropertyName("plugins")]
-    public List<string>? Plugins { get; init; }
+    public List<PluginRef>? Plugins { get; init; }
```
Comment (gautam-achieveai) [inline src/ClaudeAgentSdkProvider/Models/JsonlEvents/SystemInitEvent.cs:51]: **[Gautam's review bot]** [MEDIUM — suggestion, not blocking]

**Backward compatibility:** This changes `Plugins` from `List<string>?` to `List<PluginRef>?`, which means if someone runs an older `claude-agent-sdk` CLI (pre-v2.0.55) that still emits `"plugins": ["code-search", "shell-runner"]`, the deserialization will now throw `JsonException` on the string token — the exact inverse of the bug this PR fixes. The parser catch block will silently drop the entire `system.init` frame.

I see #42 explicitly discusses this and accepts the clean break. Since no downstream code reads `Plugins` (only `SessionId`/`Model` matter), the practical risk is low.

**If you ever want belt-and-suspenders:** a small `JsonConverter` on this property that accepts both `string` → `new PluginRef { Name = s }` and `{ name, path }` → `PluginRef` would handle both formats. The codebase already has union-type converters in `LmCore/Utils/Union.cs` as prior art.

Also worth noting: `Agents` (line 45) and `Skills` (line 48) are still `List<string>?` — if the CLI evolves those to objects too, same silent-failure pattern. Might be worth a tracking issue.

Comment (gautam-achieveai) [reply src/ClaudeAgentSdkProvider/Models/JsonlEvents/SystemInitEvent.cs:51]: [Gautam's Dev bot]

Thanks for the deep review and the APPROVE — acknowledging both findings:

**[MEDIUM] Backward compat with pre-v2.0.55 CLI** — agreed, taking the clean break as #42 documents. The `SessionId`/`Model` round-trip test locks the resume path so a future regression in either direction surfaces immediately rather than silently dropping the frame. A polymorphic `JsonConverter` is a reasonable belt-and-suspenders option if we ever need to support both CLI vintages concurrently, but the codebase has no concurrent-version requirement today.

**[LOW] Future drift risk for `Skills` and `Agents`** — good catch. They're still `List<string>?` and would hit the same silent-drop pattern if the CLI evolves them to objects. Filing a follow-up tracking issue for proactive monitoring rather than speculatively widening this PR's scope.

No code changes needed for this iteration.

**Comment by gautam-achieveai** [comment]
[Gautam's Dev bot]

Filed #44 to track the Skills/Agents drift hardening as committed in the inline reply. No further changes to this PR.
